packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -53,16 +100,43 @@ ContextMenu.displayName = CONTEXT_MENU_NAME;
  * -----------------------------------------------------------------------------------------------*/
 
 const TRIGGER_NAME = 'ContextMenuTrigger';
-const TRIGGER_DEFAULT_TAG = 'span';
 
-type ContextMenuTriggerOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ContextMenuTriggerOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof ContextMenuAreaTrigger>,
+  Polymorphic.OwnProps<typeof MenuPrimitive.SubTrigger>
+>;
 type ContextMenuTriggerPrimitive = Polymorphic.ForwardRefComponent<
-  typeof TRIGGER_DEFAULT_TAG,
+  Polymorphic.IntrinsicElement<typeof ContextMenuAreaTrigger>,
   ContextMenuTriggerOwnProps
 >;
 
 const ContextMenuTrigger = React.forwardRef((props, forwardedRef) => {
-  const { as = TRIGGER_DEFAULT_TAG, ...triggerProps } = props;
+  const { as, ...triggerProps } = props;
+  const isSubmenu = React.useContext(NestedMenuContext);
+
+  return isSubmenu ? (
+    <MenuPrimitive.SubTrigger
+      {...triggerProps}
+      as={as as Polymorphic.IntrinsicElement<typeof ContextMenuAreaTrigger>}
```

Comment:
Perhaps I'm missing a trick but I'm having to cast because of the merge between the differing IntrinsicElements (`span`, `button`) is there a better way?